types/types.go

```diff
@@ -144,6 +145,10 @@ type ImageDestination interface {
 	// ShouldCompressLayers returns true iff it is desirable to compress layer blobs written to this destination.
 	ShouldCompressLayers() bool
 
+	// CopyForeignLayers returns true iff foreign layers in manifest should be actually
+	// uploaded to the image destination, false otherwise.
+	CopyForeignLayers() bool
```

Comment:
Perhaps `ShouldCopyForeignLayers` or `SupportsForeignLayers` or something; this reads like a command to copy them.